### ISSUE TITLE
PluginAPI Nested WrapSelectors not possible

### ISSUE THREAD

**Issue by mathis-m** [swagger-api/swagger-ui#7157]
When using the plugin api it is not possible to wrap a selector twice to create a more nested dependency chain of plugins.
Whis is relevant since this takes us the possibility to wrap selectors that are already wrapped by e.g. oas3 core plugin.
To reproduce register multiple plugins that wrap the same selector.

Expected:
`window.ui.getSystem().probeSelectors.probe()` should call baseProbe => wrap1 => wrap2
Actual:
returns baseProbe => wrap2
```js
const pluginProbeBase = {
  statePlugins: {
    probe: {
      selectors: {
        probe: () => {
          console.log("[probeBase] Base selector called")
          return "base"
        }
      }
    }
  }
}
const presetProbe = [pluginProbeBase]
const probeWrap1 = {
  statePlugins: {
    probe: {
      wrapSelectors: {
        probe: (oriSelector, system) => (state, ...args) => { 
          const baseResult = oriSelector(state, ...args)
          if(baseResult === "base") {
            console.log("[probeWrap1] Recieved base result")
          } else {
            console.log(`[probeWrap1] Recieved unknown result ${baseResult}`)
          }
          return "wrap1"
        }
      }
    }
  }
}
const probeWrap2 = {
  statePlugins: {
    probe: {
      wrapSelectors: {
        probe: (oriSelector, system) => (state, ...args) => { 
          const wrapResult = oriSelector(state, ...args)
          if(wrapResult === "wrap1") {
            console.log("[probeWrap2] Recieved wrap1 result")
          } else {
            console.log(`[probeWrap2] Recieved unknown result ${wrapResult}`)
          }
          return "wrap2"
        }
      }
    }
  }
}

const ui = SwaggerUIBundle({
  url: "https://petstore.swagger.io/v2/swagger.json",
  dom_id: '#swagger-ui',
  deepLinking: true,
  presets: [
    SwaggerUIBundle.presets.apis,
    SwaggerUIStandalonePreset,
    presetProbe
  ],
  plugins: [
    SwaggerUIBundle.plugins.DownloadUrl,
    probeWrap1,
    probeWrap2
  ],
  layout: "StandaloneLayout"
});
```
_Originally posted by @mathis-m in https://github.com/swagger-api/swagger-ui/issues/7155#issuecomment-816265056_

**Comment by mathis-m** [comment]
@tim-lai I have found the issue, the following extendSystem snipped should be run for wrapSelectors too!
https://github.com/swagger-api/swagger-ui/blob/cc408812fc927e265da158bf68239530740ab4cc/src/core/system.js#L385-L411

**Comment by mathis-m** [comment]
Workaround can be found here https://github.com/swagger-api/swagger-ui/issues/7155#issuecomment-816575055

**Comment by tim-lai** [comment]
@mathis-m Thanks for researching! Fyi, one concern I have is around precedence, but it is not a new issue, and will likely need to continue to rely on the order of loading plugins.

**Comment by mathis-m** [comment]
@tim-lai Yeah of cause. Shall I prepare a PR for nesting multiple wrapSelectors according to registration order?

**Comment by tim-lai** [comment]
> @tim-lai Yeah of cause. Shall I prepare a PR for nesting multiple wrapSelectors according to registration order?

Sure! 😉

**Comment by tim-lai** [comment]
@mathis-m fyi, I did a quick experiment if one can just mostly copy `// Account for wrapActions` for `wrapSelectors`, but it didn't work, and it generated duplicate functions. I am not planning on working on this issue in the foreseeable future.

**Comment by tim-lai** [comment]
fixed in #7304.
